cometd-java/cometd-java-server/cometd-java-server-common/src/main/java/org/cometd/server/BayeuxServerImpl.java

```diff
@@ -124,6 +131,11 @@ protected void doStart() throws Exception {
         _validation = getOption(VALIDATE_MESSAGE_FIELDS_OPTION, true);
         _broadcastToPublisher = getOption(BROADCAST_TO_PUBLISHER_OPTION, true);
 
+        long sweepThreads = getOption(SWEEP_THREADS_OPTION, 2);
+        if (sweepThreads < 2)
+            sweepThreads = 2;
+        _sweepThreads = (int)Math.min(sweepThreads, Runtime.getRuntime().availableProcessors());
+
         super.doStart();
```

Comment:
Please move this above, just after the last call to `addBean()`.

Reply:
Done.

Reply:
I've moved the equivalent `sweepPeriod` code as well.